File: exercises/go/practice/space-age/space_age_test.go
package space

import (
	"math"
	"testing"
)

func TestAge(t *testing.T) {
	const precision = 0.01
	for _, tc := range testCases {
		actual := Age(tc.seconds, tc.planet)
		if math.Abs(actual-tc.expected) > precision {
			t.Fatalf("FAIL: %s\nExpected: %#v\nActual: %#v", tc.description, tc.expected, actual)
		}
		t.Logf("PASS: %s", tc.description)
	}
}

func BenchmarkAge(b *testing.B) {
	for i := 0; i < b.N; i++ {
		for _, tc := range testCases {
			Age(tc.seconds, tc.planet)
		}
	}
}

var testCases = []struct {
	description string
	seconds     float64
	expected    float64
	planet      Planet
}{
	{
		description: "age on Earth",
		seconds:     1000000000,
		expected:    31.69,
		planet:      "Earth",
	},
	{
		description: "age on Mercury",
		seconds:     2134835688,
		expected:    280.88,
		planet:      "Mercury",
	},
	{
		description: "age on Venus",
		seconds:     189839836,
		expected:    9.78,
		planet:      "Venus",
	},
	{
		description: "age on Mars",
		seconds:     2329871239,
		expected:    39.25,
		planet:      "Mars",
	},
	{
		description: "age on Jupiter",
		seconds:     901876382,
		expected:    2.41,
		planet:      "Jupiter",
	},
	{
		description: "age on Saturn",
		seconds:     3000000000,
		expected:    3.23,
		planet:      "Saturn",
	},
	{
		description: "age on Uranus",
		seconds:     3210123456,
		expected:    1.21,
		planet:      "Uranus",
	},
	{
		description: "age on Neptune",
		seconds:     8210123456,
		expected:    1.58,
		planet:      "Neptune",
	},
}

File: exercises/typescript/practice/space-age/tests/space-age.test.ts
import { describe, it, expect } from 'vitest';
import { SpaceAge } from '../src/space-age';

function assertInDelta(actual: number, expected: number) {
  const delta = 0.01;
  expect(Math.abs(actual - expected)).toBeLessThanOrEqual(delta);
}

describe('Space Age', () => {
  it('age on Earth', () => {
    const age = new SpaceAge(1000000000);
    assertInDelta(age.onEarth(), 31.69);
  });

  it('age on Mercury', () => {
    const age = new SpaceAge(2134835688);
    assertInDelta(age.onMercury(), 280.88);
  });

  it('age on Venus', () => {
    const age = new SpaceAge(189839836);
    assertInDelta(age.onVenus(), 9.78);
  });

  it('age on Mars', () => {
    const age = new SpaceAge(2329871239);
    assertInDelta(age.onMars(), 39.25);
  });

  it('age on Jupiter', () => {
    const age = new SpaceAge(901876382);
    assertInDelta(age.onJupiter(), 2.41);
  });

  it('age on Saturn', () => {
    const age = new SpaceAge(3000000000);
    assertInDelta(age.onSaturn(), 3.23);
  });

  it('age on Uranus', () => {
    const age = new SpaceAge(3210123456);
    assertInDelta(age.onUranus(), 1.21);
  });

  it('age on Neptune', () => {
    const age = new SpaceAge(8210123456);
    assertInDelta(age.onNeptune(), 1.58);
  });
});